### PR TITLE
Remove source when deiniting annotation manager

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes https://github.com/mapbox/mapbox-maps-ios/issues/616

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

The wrong identifier was being passed into `removeSource(withId:)` (`layerId` instead of `sourceId`). This change fixes the identifier.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,8 +26,9 @@ Mapbox welcomes participation and contributions from everyone.
 * Fix volatile tiles disappearing on "not modified" response ([#628](https://github.com/mapbox/mapbox-maps-ios/pull/628))
 * Fix crash in MapboxMap.clearData() ([#628](https://github.com/mapbox/mapbox-maps-ios/pull/628))
 * Trigger map redraw when feature state changes ([#628](https://github.com/mapbox/mapbox-maps-ios/pull/628))
-* Do not start background task if telemetry collection is disabled ([#](https://github.com/mapbox/mapbox-maps-ios/pull/))
-* Fix KVC decoding for iOS 15 ([#](https://github.com/mapbox/mapbox-maps-ios/pull/))
+* Do not start background task if telemetry collection is disabled ([#631](https://github.com/mapbox/mapbox-maps-ios/pull/631))
+* Fix KVC decoding for iOS 15 ([#631](https://github.com/mapbox/mapbox-maps-ios/pull/631))
```
Comment (macdrevx) [inline CHANGELOG.md:30]: Thank you. I think there's another one somewhere

**Review by macdrevx — APPROVED** [review]

**Comment by macdrevx** [comment]
@nishant-karajgikar one note here — this solution may not be sufficiently robust. In https://developer.apple.com/videos/play/wwdc2021/10216/, Apple warns against relying on side-effects of deinit. So we could run into issues if the new annotation manager has the same ID as the first one.

**Comment by nishant-karajgikar** [comment]
> @nishant-karajgikar one note here — this solution may not be sufficiently robust. In https://developer.apple.com/videos/play/wwdc2021/10216/, Apple warns against relying on side-effects of deinit. So we could run into issues if the new annotation manager has the same ID as the first one.

Oh good call. I've gone ahead and ticketed this out.

File: CHANGELOG.md (hunk 1 above)
Comment (nishant-karajgikar) [reply CHANGELOG.md:30]: Got that one too.

**Pull request merged**